### ISSUE TITLE
Firebase Is Not Working With Expo Cli 2.20.2 Generated Project

### ISSUE THREAD

**Issue by joedinar** [expo/expo#4677]
### Environment

  Expo CLI 2.20.2 environment info:
    System:
      OS: macOS 10.14.5
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 11.12.0 - /usr/local/bin/node
      npm: 6.9.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.3 AI-182.5107.16.33.5264788
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmGlobalPackages:
      expo-cli: 2.20.2

Target iOS

### Steps to Reproduce

Created a new project with expo init

ran npm install --save firebase

Setup a connection to Firestore by following expo's guide https://docs.expo.io/versions/latest/guides/using-firebase/

Got the following error `Unable to resolve module @react-native-community/async-storage from ../node_modules/@firebase/app/dist/index.rn.cjs.js: Module @react-native-community/async-storage does not exist in the Haste module map`

Followed the suggested steps from the error report, and still get the same error.

### Expected Behavior

App to load and be able to access my Firestore database. I have other expo projects that were created prior to upgrading to expo-cli 2.20.2, that work with Firebase.

### Actual Behavior

Received the following error: `Unable to resolve module @react-native-community/async-storage from ../node_modules/@firebase/app/dist/index.rn.cjs.js: Module @react-native-community/async-storage does not exist in the Haste module map`

**Comment by TMcLoone** [comment]
Having this same issue also. Any ideas on how to work around it?

**Comment by joedinar** [comment]
Aside from rolling back to a previous version of expo and recreating the project I tried everything I could think of with no luck. I am currently putting the firebase integration on the side while I finish up other aspects of the app and hoping a fix is released by the time I come back around to it. Otherwise I'm most likely going to roll back.

**Comment by cruzach** [comment]
I think this is a problem with Firebase, not Expo. [Relevant issue here](https://github.com/firebase/firebase-js-sdk/issues/1898)

**Comment by joedinar** [comment]
@cruzach I think you are right. I used expo init to create a new project with expo-cli 2.20.2 and installed firebase@5.9.2 instead of the latest. The project booted up just fine and everything seems to be working and I have a connection to the DB.

For the original project I had the issue with I ran npm uninstall firebase for the initial project I was using, deleted the package-lock and node_modules, installed firebase@5.9.2 and reran npm install and still had the same issue. 

I then copied all folders from original project into the newly created one and everything worked fine. Confirming the issue is with Firebase's latest release. @TMcLoone 

@cruzach Thank you for pointing that out you saved me a lot of headaches. Closing the issue.
